### DISCUSSION TITLE
Advanced Group Permissions and Restrictive Access Controls on Documents

### DISCUSSION THREAD

**Discussion by dariofflow** [outline/outline#6846] [Ideas]
Dear Outline Team,

I've come across a limitation in the current permission system that could be improved for better document management within teams. While the existing 'share' option on individual documents allows for adding users, it does not support the complexity required for dynamic team structures, especially in a corporate setting.

### Feature Enhancement Request:

1. **Group Permissions:** The ability to add entire groups to the permissions setting of individual documents. This would streamline the process of managing access for teams rather than adding users one by one.
2. **Revoking Access:** Currently, if a group has access to a collection, they inherently have access to all documents within. We need the functionality to selectively revoke access to specific documents for a group that otherwise has access to the entire collection.

### Practical Scenario:
For example, if the 'Architects' group has access to a collection, I want to restrict their access to a specific document within that collection. This level of control is crucial for maintaining strict information boundaries and ensuring that sensitive information is only accessible to the right individuals or groups.

### Business Impact:
The addition of these features would greatly enhance security and efficiency in managing team access to documents. It provides a tailored experience that supports the diverse access needs within larger teams and protects sensitive data more effectively.

I believe that these proposed enhancements align with the collaborative and flexible spirit of Outline, and would significantly benefit organizations looking for precise control over their knowledge bases.

Thank you for considering this request and for your continuous efforts to improve the Outline experience.

Warm regards,
Dario

**Comment by tommoor** [comment]
👍 Please see existing issue here:
https://github.com/outline/outline/issues/6639

**Reply by dariofflow** [reply]
Dear @tommoor 

Thank you for directing me to the existing issue #6639. I have reviewed it and noticed that while it covers the addition of group permissions to individual documents, it does not address the feature of revoking access from specific documents for groups that have overarching access to a collection.

This second aspect is crucial for granular access control within an organization, as it allows for sensitive documents to be selectively hidden from groups that have general access to a collection. This is particularly important for ensuring that confidential information remains secure and only accessible to authorized users or groups.

Would it be possible to address this feature separately or expand the existing issue to include this functionality? The ability to revoke access at a document level for groups would complete the suite of permissions controls that we need for effective document management in a collaborative environment.

Thank you again for your attention to this matter.

Warm regards,
Dario

**Reply by tommoor** [reply]
While I'll never say never, it's highly unlikely we'll build this feature – permissions in Outline are designed to be additive for user clarity and performance reasons.

I'm being clear on this as if it's a dealbreaker feature for you then you should consider another platform.
